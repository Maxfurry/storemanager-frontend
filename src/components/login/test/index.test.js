import React from 'react';
import { mount } from 'enzyme';
import { Login } from '..';

describe('LoginForm', () => {
  it('should match snapshot', () => {
    const wrapper = mount(<Login />);
    expect(wrapper).toMatchSnapshot();
  });
});

it('should change the state after change the input value', () => {
  const name = 'fury';
  const wrapper = mount(<Login />);
  wrapper.find('input[type="text"]').simulate('change', {
    target: { name }
  });
  wrapper.update();
  expect(wrapper.find('input[type="text"]').prop('name')).toContain('name');
  expect(wrapper.find('input[type="password"]').prop('name')).toContain('password');
});

it('should simulate on submit event ', () => {
  const props = {
    userLogin: () => {}
  };
  const wrapper = mount(<Login {... props} />);
  wrapper.find('form').simulate('submit', {
    preventDefault: jest.fn()
  });
  expect(wrapper.find('form')).toBeDefined();
});
