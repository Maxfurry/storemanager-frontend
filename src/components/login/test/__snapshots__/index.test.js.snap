// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match snapshot 1`] = `
<div
  className="login-form"
>
  <Grid
    style={
      Object {
        "height": "100%",
      }
    }
    textAlign="center"
    verticalAlign="middle"
  >
    <GridColumn
      style={
        Object {
          "maxWidth": 450,
        }
      }
    >
      <Form
        as="form"
        size="large"
      >
        <Segment
          stacked={true}
        >
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="user"
            iconPosition="left"
            placeholder="Username"
          />
          <p>
             Password 
          </p>
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="lock"
            iconPosition="left"
            placeholder="Password"
            type="password"
          />
          <Button
            as="button"
            className="loginButton"
            size="large"
          >
            Login
          </Button>
        </Segment>
      </Form>
    </GridColumn>
  </Grid>
</div>
`;
